include/seqan3/io/alignment_file/detail.hpp

```diff
@@ -238,106 +251,65 @@ std::string get_cigar_string(alignment_type && alignment,
+ * \tparam query_seq_type  Must model std::ranges::forward_range. The value_type must
+ *                         be equality comparable to seqan3::gap.
+ * \param  ref_seq         The reference sequence to compare against the query sequence.
+ * \param  query_seq       The query sequence to build the CIGAR string for.
  * \param  query_start_pos The start position of the alignment in the query
  *                         sequence indicating soft-clipping.
  * \param  query_end_pos   The end position of the alignment in the query
  *                         sequence indicating soft-clipping.
  * \param  extended_cigar  Whether to print the extended cigar alphabet or not. See CIGAR operation.
- * \returns An std::vector<seqan3::cigar> representing the alignment.
+ * \returns An std::string representing the alignment as a CIGAR string.
  *
  * \details
  *
  * \attention Note that CIGAR elements (respectively by their
- *            CIGAR operation) are always related to on one of the two sequences
+ *            CIGAR operation) are always related to one of the two sequences
  *            in a pairwise alignment. In this case, the resulting cigar_vector
  *            is based on sequence at the second position of the \p alignment pair,
  *            namely the query sequence.
  *
- * ### Example:
+ * ### Theoretical Example:
  *
- * Given the following alignment reference sequence on top and the query sequence at
- * the bottom:
+ * The following alignment reference sequence on top and the query sequence at
+ * the bottom.
  * ```
  * ATGG--CGTAGAGC
  * |||X  |||X|  |
  * ATGCCCCGTTG--C
  * ```
- * In this case, the function seqan3::detail::get_cigar_vector will return
- * the following cigar vector: "[('M',4),('I',2),('M',5),('D',2),('M',1)]".
- * The extended cigar string would look like this: "[('=',3)('X',1)('I',2)('=',3)('X',1)('=',1)('D',2)('=',1)]".
- *
- * \include test/snippet/io/alignment_file/get_cigar_vector.cpp
+ * In this case, the function seqan3::detail::get_cigar_string will return
+ * the following cigar string when printed: "4M2I5M2D1M". The extended cigar
+ * string would look like this: "3=1X2I3=1X1=2D1=".
  *
  * \sa seqan3::aligned_sequence
  */
-template <tuple_like alignment_type>
+template <std::ranges::forward_range ref_seq_type, std::ranges::forward_range query_seq_type>
 //!\cond
-    requires std::tuple_size_v<remove_cvref_t<alignment_type>> == 2
+    requires std::detail::weakly_equality_comparable_with<gap, reference_t<ref_seq_type>> &&
+             std::detail::weakly_equality_comparable_with<gap, reference_t<query_seq_type>>
 //!\endcond
-std::vector<cigar> get_cigar_vector(alignment_type && alignment,
-                                    uint32_t const query_start_pos = 0,
-                                    uint32_t const query_end_pos = 0,
-                                    bool const extended_cigar = false)
+std::string get_cigar_string(ref_seq_type && ref_seq,
```

Comment:
I know you are not in favour of this whole design but I would like to avoid too many naming changes in this already big PR. I can refactor all the namings in a later PR, if that is fine?

Reply:
I agree, there is a lot of things that are going on. A lot of things we need to revisit for file design anyway. So rather have a dedicated discussion and plan for this in the future. Let's not change it now then, because I feel more will be happening